File: app/src/main/java/com/hussain/podcastapp/adapter/EpisodeAdapter.java
package com.hussain.podcastapp.adapter;

import android.content.Context;
import android.graphics.Bitmap;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.airbnb.lottie.ImageAssetDelegate;
import com.airbnb.lottie.LottieDrawable;
import com.airbnb.lottie.LottieImageAsset;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.hussain.podcastapp.R;
import com.hussain.podcastapp.model.Item;
import com.hussain.podcastapp.utils.GlideApp;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class EpisodeAdapter extends RecyclerView.Adapter<EpisodeAdapter.EpisodeViewHolder> {

    private List<Item> mItemList;
    private EpisodeClickListener mClickListener;
    private Context mContext;

    public EpisodeAdapter(List<Item> items, EpisodeClickListener clickListener) {
        this.mItemList = items;
        this.mClickListener = clickListener;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public EpisodeViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        mContext = viewGroup.getContext();
        View view = LayoutInflater.from(mContext).inflate(R.layout.layout_episode_item, viewGroup, false);
        return new EpisodeViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull EpisodeViewHolder episodeViewHolder, int i) {
        final Item item = mItemList.get(i);
        episodeViewHolder.mTitle.setText(item.getTitle());
        episodeViewHolder.mSummary.setText(Html.fromHtml(item.getSummary()));
        LottieDrawable drawable = new LottieDrawable();
        drawable.setImageAssetDelegate(new ImageAssetDelegate() {
            @Override
            public Bitmap fetchBitmap(LottieImageAsset asset) {
                return null;
            }
        });
        GlideApp.with(mContext)
                .load(item.getImage())
                .centerCrop()
                .placeholder(R.drawable.placeholder_square)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(episodeViewHolder.mThumb);
        episodeViewHolder.itemView.setOnClickListener(view -> {
            if (mClickListener != null) {
                mClickListener.onEpisodeClick(item, i, episodeViewHolder.mThumb);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mItemList.size();
    }

    public interface EpisodeClickListener {
        void onEpisodeClick(Item item, int position, ImageView imageView);
    }

    class EpisodeViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.ivThumb)
        ImageView mThumb;
        @BindView(R.id.tvTitle)
        TextView mTitle;
        @BindView(R.id.tvSummary)
        TextView mSummary;

        EpisodeViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/hussain/podcastapp/adapter/EpisodeAdapter.kt
package com.hussain.podcastapp.adapter

import android.content.Context
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.annotation.NonNull
import androidx.recyclerview.widget.RecyclerView
import com.airbnb.lottie.LottieDrawable
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.hussain.podcastapp.R
import com.hussain.podcastapp.model.Item
import com.hussain.podcastapp.utils.GlideApp
import kotlinx.android.synthetic.main.layout_episode_item.view.*

class EpisodeAdapter(private val mItemList: List<Item>, private val mClickListener: EpisodeClickListener?) : RecyclerView.Adapter<EpisodeAdapter.EpisodeViewHolder>() {
    private var mContext: Context? = null

    init {
        notifyDataSetChanged()
    }

    @NonNull
    override fun onCreateViewHolder(@NonNull viewGroup: ViewGroup, i: Int): EpisodeViewHolder {
        mContext = viewGroup.context
        val view = LayoutInflater.from(mContext).inflate(R.layout.layout_episode_item, viewGroup, false)
        return EpisodeViewHolder(view)
    }

    override fun onBindViewHolder(@NonNull episodeViewHolder: EpisodeViewHolder, i: Int) {
        val item = mItemList[i]
        episodeViewHolder.mTitle.text = item.title
        episodeViewHolder.mSummary.text = Html.fromHtml(item.summary)
        val drawable = LottieDrawable()
        drawable.setImageAssetDelegate { null }
        GlideApp.with(mContext!!)
                .load(item.image)
                .centerCrop()
                .placeholder(R.drawable.placeholder_square)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(episodeViewHolder.mThumb)
        episodeViewHolder.itemView.setOnClickListener { view ->
            mClickListener?.onEpisodeClick(item, i, episodeViewHolder.mThumb)
        }
    }

    override fun getItemCount(): Int {
        return mItemList.size
    }

    interface EpisodeClickListener {
        fun onEpisodeClick(item: Item, position: Int, imageView: ImageView?)
    }

    class EpisodeViewHolder(@NonNull itemView: View) : RecyclerView.ViewHolder(itemView) {

        val mThumb = itemView.ivThumb!!
        var mTitle = itemView.tvTitle!!
        var mSummary = itemView.tvSummary!!
    }
}
